app/src/app/utils/GeometryWorker/geometryWorker.ts

```diff
@@ -247,20 +236,27 @@ const GeometryWorker: GeometryWorkerClass = {
     }
 
     if (fast) {
+      const minimumDistance = minBuffer ?? CENTROID_BUFFER_KM
+      // convert 30pixels
       Object.entries(this.previousCentroids).forEach(([zone, centroid]) => {
         if (!centroid.properties?.id || !centroid.properties.zone) return;
         const props = centroid.properties;
-        const currZone = this.activeGeometries[props.id].properties.zone;
-        if (currZone === null || (currZone !== props.zone && activeZones.includes(currZone)))
-          return;
+        const currZone = this.activeGeometries[props.id]?.properties?.zone;
+        if (!currZone || currZone !== props.zone && activeZones.includes(currZone)) return;
```

Comment:
PP: for complex logical statements like these, adding a comment or grouping terms into variables would be a bit clearer. Also my understanding of the code here is a bit iffy so pls rename this variable if you implement this idea!

```suggestion
        const currZoneIsPainted = currZone !== props.zone && activeZones.includes(currZone)
        if (!currZone || currZoneIsPainted) return;
```

Reply:
Agreed, good call